docs/standard/exceptions/best-practices-for-exceptions.md

```diff
@@ -249,6 +194,122 @@ File name: 'C:\temp\file.txt'
+
+If you're implementing an asynchronous method, call <xref:System.Threading.CancellationToken.ThrowIfCancellationRequested?displayProperty=nameWithType> instead of checking if cancellation was requested and then constructing and throwing <xref:System.OperationCanceledException>. For more information, see [CA2250](../../fundamentals/code-analysis/quality-rules/ca2250.md).
+
+### Include a localized string message
+
+The error message the user sees is derived from the <xref:System.Exception.Message?displayProperty=nameWithType> property of the exception that was thrown, and not from the name of the exception class. Typically, you assign a value to the <xref:System.Exception.Message?displayProperty=nameWithType> property by passing the message string to the `message` argument of an [Exception constructor](xref:System.Exception.%23ctor%2A).
+
+For localized applications, you should provide a localized message string for every exception that your application can throw. You use resource files to provide localized error messages. For information on localizing applications and retrieving localized strings, see the following articles:
+
+- [How to: Create user-defined exceptions with localized exception messages](how-to-create-localized-exception-messages.md)
+- [Resources in .NET apps](../../core/extensions/resources.md)
+- <xref:System.Resources.ResourceManager?displayProperty=nameWithType>
+
+### Use proper grammar
+
+Write clear sentences and include ending punctuation. Each sentence in the string assigned to the <xref:System.Exception.Message?displayProperty=nameWithType> property should end in a period. For example, "The log table has overflowed." uses correct grammar and punctuation.
+
+### Place throw statements well
+
+Place throw statements where the stack trace will be helpful. The stack trace begins at the statement where the exception is thrown and ends at the `catch` statement that catches the exception.
+
+### Rethrow to preserve stack details
+
+Once an exception is thrown, part of the information it carries is the stack trace. The stack trace is a list of the method call hierarchy that starts with the method that throws the exception and ends with the method that catches the exception. If an exception is rethrown by specifying the exception in the `throw` statement, the stack trace is restarted at the current method and the list of method calls between the original method that threw the exception and the current method is lost. To keep the original stack trace information with the exception, use the `throw` statement without specifying the exception.
```

Comment:
> To keep the original stack trace information with the exception, use the `throw` statement without specifying the exception.

This is only relevant if you're inside of the catch block that's caught that exception instance.